src/assets/wise5/services/teacherDataService.ts

```diff
@@ -185,7 +185,28 @@ export class TeacherDataService extends DataService {
 
   getAllRelatedComponents(node: Node): any {
     const components = node.getNodeIdComponentIds();
-    return components.concat(this.getConnectedComponentsWithRequiredStudentData(components));
+    return [
+      ...components,
+      ...this.getShowMyWorkStudentData(components),
+      ...this.getConnectedComponentsWithRequiredStudentData(components)
+    ];
+  }
+
+  getShowMyWorkStudentData(components: any[]): any[] {
```

Comment:
What if this function takes in the Node object? Would it simplify the code?

```
(pseudocode)
return node.components
   .filter( component.type === ShowMyWork )
    .map( make nodeId componentId object );
```

Also, make this function private?